test/test_sparse.py

```diff
@@ -3177,12 +3195,53 @@ def assign_to(a):
         self.assertRaises(TypeError, assign_to)
 
 
+class TestSparseComplex(TestSparse):
+
+    value_dtype = torch.complex128
+
+    def test_dsmm(self):
+        self.skipTest("torch.dsmm skipped! Not implemented yet!")
+
+    def test_hsmm(self):
+        self.skipTest("torch.hsmm skipped! Not implemented yet!")
+
+    def test_print(self):
+        self.skipTest("test_print skipped! arange for complex data not implemented yet!")
+
+    def test_pickle(self):
+        self.skipTest("test_pickle skipped! arange for complex data not implemented yet!")
+
+
+class TestUncoalescedSparseComplex(TestSparseComplex):
+
+    def setUp(self):
+        super(TestUncoalescedSparseComplex, self).setUp()
+        self.is_uncoalesced = True
+
+
 class TestUncoalescedSparse(TestSparse):
     def setUp(self):
         super(TestUncoalescedSparse, self).setUp()
         self.is_uncoalesced = True
 
 
+@unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+class TestCudaSparseComplex(TestSparseComplex):
+
+    def setUp(self):
+        super(TestCudaSparseComplex, self).setUp()
+        self.is_cuda = True
+        self.device = 'cuda'
+
+
+@unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+class TestCudaUncoalescedSparseComplex(TestCudaSparseComplex):
```

Comment:
We should really modernize test_sparse.py to use the device generic test framework (and OpInfos) and simplify this variant instantiation process.

Reply:
I'm working on this, I think this task really worth. Using the new generic test framework I've found out some issues like misuses of the API and some memory leaks.